File: android/app/src/main/java/com/github/makewheels/video2022/data/model/Video.kt
package com.github.makewheels.video2022.data.model

data class VideoItem(
    val id: String,
    val watchId: String?,
    val title: String?,
    val description: String?,
    val status: String?,
    val visibility: String?,
    val watchCount: Int?,
    val duration: Long?,
    val createTime: String?,
    val createTimeString: String?,
    val watchUrl: String?,
    val shortUrl: String?,
    val type: String?,
    val coverUrl: String?,
    val youtubePublishTimeString: String?,
    val uploaderName: String? = null,
    val uploaderAvatarUrl: String? = null,
    val uploaderId: String? = null
)

data class VideoListResponse(
    val list: List<VideoItem>,
    val total: Long
)

data class CreateVideoRequest(
    val videoType: String,
    val rawFilename: String? = null,
    val youtubeUrl: String? = null,
    val size: Long? = null,
    val ttl: String = "PERMANENT"
)

data class CreateVideoResponse(
    val watchId: String,
    val shortUrl: String,
    val videoId: String,
    val watchUrl: String,
    val fileId: String
)

data class UpdateVideoInfoRequest(
    val id: String,
    val title: String? = null,
    val description: String? = null,
    val visibility: String? = null
)

data class VideoStatus(
    val videoId: String,
    val status: String,
    val isReady: Boolean
)

data class UpdateWatchSettingsRequest(
    val videoId: String,
    val showUploadTime: Boolean? = null,
    val showWatchCount: Boolean? = null
)

File: ios/VideoApp/Sources/Models/Video.swift
import Foundation

struct VideoItem: Decodable, Identifiable {
    let id: String
    let watchId: String?
    let title: String?
    let description: String?
    let status: String?
    let visibility: String?
    let watchCount: Int?
    let duration: Int?
    let createTime: String?
    let createTimeString: String?
    let watchUrl: String?
    let shortUrl: String?
    let type: String?
    let coverUrl: String?
    let youtubePublishTimeString: String?
    let uploaderName: String?
    let uploaderAvatarUrl: String?
    let uploaderId: String?
    let tags: [String]?
    let category: String?
}

struct VideoListResponse: Decodable {
    let list: [VideoItem]
    let total: Int
}

struct CreateVideoRequest: Encodable {
    let videoType: String
    let rawFilename: String?
    let youtubeUrl: String?
    let size: Int64?
    let ttl: String
    
    init(videoType: String, rawFilename: String? = nil, youtubeUrl: String? = nil, size: Int64? = nil, ttl: String = "PERMANENT") {
        self.videoType = videoType
        self.rawFilename = rawFilename
        self.youtubeUrl = youtubeUrl
        self.size = size
        self.ttl = ttl
    }
}

struct CreateVideoResponse: Decodable {
    let watchId: String
    let shortUrl: String
    let videoId: String
    let watchUrl: String
    let fileId: String
}

struct UpdateVideoInfoRequest: Encodable {
    let id: String
    let title: String?
    let description: String?
    let visibility: String?
    var tags: [String]?
    var category: String?
}
